### PR TITLE
V13

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: src/geode/inspector/criterion/intersections/model_intersections.cpp [hunk 1]
```diff
@@ -94,31 +95,18 @@ namespace
...
-            if( common_vertices.size() == 3
-                || triangles_intersect(
-                    t1_id, t2_id, t1_vertices, t2_vertices, common_vertices ) )
-            {
-                intersecting_triangles_.emplace_back( t1_id, t2_id );
-            }
-        }
+        virtual ~ModelSurfacesIntersectionBase() = default;
+
+        virtual bool operator()(
```
Comment (BotellaA) [inline src/geode/inspector/criterion/intersections/model_intersections.cpp:100]: The polymorphism is not really needed. We could remove the destructor and the virtual operator I think and directly use the one from the children

File: src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp [hunk 2]
```diff
@@ -43,43 +43,91 @@
...
-        void operator()( geode::index_t triangle_id, geode::index_t edge_id )
-        {
-            if( edge_intersects_triangle( triangle_id, edge_id ) )
-            {
-                intersecting_elements_.emplace_back( triangle_id, edge_id );
-            }
-        }
+        virtual ~TriangleEdgeIntersectionBase() = default;
+
+        virtual bool operator()(
```
Comment (BotellaA) [inline src/geode/inspector/criterion/intersections/surface_curve_intersections.cpp:58]: Same here

File: src/geode/inspector/criterion/intersections/surface_intersections.cpp [hunk 3]
```diff
@@ -58,40 +58,37 @@ namespace
...
-            if( common_points.size() == 3
-                || triangles_intersect(
-                    t1_id, t2_id, t1_vertices, t2_vertices, common_points ) )
-            {
-                intersecting_triangles_.emplace_back( t1_id, t2_id );
-            }
-        }
+        virtual ~TriangleTriangleIntersectionBase() = default;
+
+        virtual bool operator()(
```
Comment (BotellaA) [inline src/geode/inspector/criterion/intersections/surface_intersections.cpp:72]: And here

Comment (panquez) [reply src/geode/inspector/criterion/intersections/surface_intersections.cpp:72]: Ok

**Review by panquez — APPROVED** [review]

**Pull request merged**